vector/src/main/java/im/vector/activity/VectorRoomActivity.java

```diff
@@ -2313,6 +2317,40 @@ public void onRequestPermissionsResult(int aRequestCode, @NonNull String[] aPerm
         }
     }
 
+    /**
+     * The user can use enter key on his soft keyboard to add a new line or to send message
+     * depending on the settings he has chosen.
+     */
+    private void manageKeyboardOptionsToSendMessage() {
+        if (PreferencesManager.useEnterKeyToSendMessage(this)) {
+            mEditText.setImeOptions(EditorInfo.IME_ACTION_DONE);
+            mEditText.setOnKeyListener(new View.OnKeyListener() {
+                @Override
+                public boolean onKey(View v, int keyCode, KeyEvent event) {
+                    if (event.getAction() == KeyEvent.ACTION_DOWN) {
+                        switch (keyCode) {
+                            case KeyEvent.KEYCODE_DPAD_CENTER:
+                            case KeyEvent.KEYCODE_ENTER:
+                                sendTextMessage();
+                                return true;
+                            default:
+                                break;
+                        }
+                    }
+                    return false;
+                }
+            });
+        } else {
+            mEditText.setImeOptions(EditorInfo.IME_ACTION_UNSPECIFIED);
+            mEditText.setInputType(InputType.TYPE_TEXT_FLAG_MULTI_LINE);
+            mEditText.setOnKeyListener(null);
+            mEditText.setSingleLine(false);
```

Comment:
Question, is multiline not compatible with send action ?

Reply:
I will test

Reply:
Unfortunately no.